lib/elixir/lib/kernel.ex

```diff
@@ -3028,31 +3028,43 @@ defmodule Kernel do
   defmacro left in right do
     in_module? = (__CALLER__.context == nil)
 
-    right = case bootstrapped?(Macro) do
-      true  -> Macro.expand(right, __CALLER__)
-      false -> right
+    expand = case bootstrapped?(Macro) do
+      true -> &Macro.expand(&1, __CALLER__)
+      false -> &(&1)
     end
 
-    case right do
+    case expand.(right) do
       [] when not in_module? ->
         false
 
-      [h | t] ->
-        in_var(in_module?, left, &in_list(&1, h, t))
+      [head | tail] = list when not in_module? ->
+        in_var(in_module?, left, &in_list(&1, head, tail, expand, list, in_module?))
+
+      [_ | _] = list when in_module? ->
+        case ensure_evaled(list, {0, []}, expand) do
+          {[head | tail], {_, []}} ->
+            in_var(in_module?, left, &in_list(&1, head, tail, expand, list, in_module?))
+          {[head | tail], {_, vars_values}} ->
```

Comment:
It is fine as is, because if you look one line below, you will get more context what this really is about.